src/components/grid/grid.tsx

```diff
@@ -1,26 +1,85 @@
 import styled, { CSSProp } from 'styled-components';
 import { up } from '../breakpoint/breakpoint';
 
+type GridItemSize = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
+
 const GRID_GAP_DESKTOP: CSSProp = '24px';
 const GRID_GAP_MOBILE: CSSProp = '12px';
 
-export const Grid = styled.div`
-  max-width: 1280px;
-  margin: 0 auto;
-  padding: 0 ${GRID_GAP_MOBILE}; // outer gap
+interface GridProps {
+  disableGap?: boolean;
+}
 
+/**
+ * This container provides a grid that has 12 columns.
+ *
+ * Grids can also be nested, as well as their gap can be disabled via prop.
+ */
+export const Grid = styled.div<GridProps>`
   display: grid;
   grid-template-columns: repeat(12, 1fr);
-  grid-column-gap: ${GRID_GAP_MOBILE};
+
+  grid-column-gap: ${(props) => (props.disableGap ? 0 : GRID_GAP_MOBILE)};
+  padding: 0 ${(props) => (props.disableGap ? 0 : GRID_GAP_MOBILE)}; // outer gap
+
+  max-width: 1280px;
+  margin: 0 auto;
   justify-items: stretch;
   align-items: start;
 
-  ${up('sm')} {
-    padding: 0 ${GRID_GAP_DESKTOP};
-    grid-column-gap: ${GRID_GAP_DESKTOP};
+  ${up('md')} {
+    grid-column-gap: ${(props) => (props.disableGap ? 0 : GRID_GAP_DESKTOP)};
+    padding: 0 ${(props) => (props.disableGap ? 0 : GRID_GAP_DESKTOP)}; // outer gap
   }
 `;
 
-export const GridItem = styled.div<{ size: number }>`
-  grid-column-start: span ${(props) => props.size};
+interface GridItemProps {
+  xs?: GridItemSize;
+  sm?: GridItemSize;
+  md?: GridItemSize;
+  lg?: GridItemSize;
+  xl?: GridItemSize;
+}
+
+/**
+ * The grid item can be placed inside the Grid component. If the amount of items in one row is larger then 12, the
+ * overflowing grid item will be placed in the next row.
+ *
+ * By default, a grid item takes the whole row (12 columns) on xs.
+ */
+export const GridItem = styled.div<GridItemProps>`
+  /* on mobile an item takes the whole row itself by default */
```

Comment:
Redundant comment, as it's already mentioned in line 48. 🤓

Reply:
removed